static/js/components/ExpandedPostDisplay.js

```diff
@@ -88,60 +98,91 @@ export default class ExpandedPostDisplay extends React.Component<*, void> {
       beginEditing,
       isModerator,
       showPostDeleteDialog,
-      showPostReportDialog
+      showPostReportDialog,
+      postDropdownMenuOpen,
+      showPostMenu,
+      hidePostMenu,
+      showPostShareMenu,
+      hidePostShareMenu,
+      postShareMenuOpen
     } = this.props
 
     return (
       <div className="post-actions">
-        <PostVotingButtons
-          post={post}
-          className="expanded"
-          toggleUpvote={toggleUpvote}
-        />
-        {SETTINGS.username === post.author_id && post.text ? (
-          <div
-            className="comment-action-button edit-post"
-            onClick={beginEditing(editPostKey(post), post)}
-          >
-            <a href="#">edit</a>
-          </div>
-        ) : null}
-        {post.num_reports ? (
-          <div className="report-count">Reports: {post.num_reports}</div>
-        ) : null}
-        {SETTINGS.username === post.author_id ? (
-          <div
-            className="comment-action-button delete-post"
-            onClick={showPostDeleteDialog}
-          >
-            <a href="#">delete</a>
-          </div>
-        ) : null}
-        {userIsAnonymous() ? null : this.postSubscriptionButton()}
-        {isModerator && !post.removed ? (
-          <div
-            className="comment-action-button remove-post"
-            onClick={this.removePost.bind(this)}
-          >
-            <a href="#">remove</a>
-          </div>
-        ) : null}
-        {isModerator && post.removed ? (
-          <div
-            className="comment-action-button approve-post"
-            onClick={this.approvePost.bind(this)}
-          >
-            <a href="#">approve</a>
-          </div>
-        ) : null}
-        {!userIsAnonymous() ? (
-          <div
-            className="comment-action-button report-post"
-            onClick={showPostReportDialog}
-          >
-            <a href="#">report</a>
+        <div className="left">
+          <PostVotingButtons
+            post={post}
+            className="expanded"
+            toggleUpvote={toggleUpvote}
+          />
+        </div>
+        <div className="right">
+          {SETTINGS.username === post.author_id && post.text ? (
+            <div
+              className="post-action edit-post"
+              onClick={beginEditing(editPostKey(post), post)}
+            >
+              <i className="material-icons edit">edit</i>
+              Edit
+            </div>
+          ) : null}
+          <div className="post-action share-action">
+            <div onClick={showPostShareMenu}>
+              <i className="material-icons reply">reply</i>
+              Share
+            </div>
+            {postShareMenuOpen ? (
+              <SharePopup
+                url={window.location.href}
```

Comment:
I don't think we want to do `window.location.href` here because it's not always the canoncial link to the post. A counterexample can be seen if you click on the "permalink" button for a comment and then click the "share" button, you get the url to the comment, not the post, which I don't think is what we want.

Reply:
makes sense, I'll write a library function for getting a post permalink